ndk/hooks/useFeedWithEose.ts

```diff
@@ -0,0 +1,39 @@
+import { type NDKEvent, type NDKFilter } from "@nostr-dev-kit/ndk";
+import { useNDK } from "ndk/NDKProvider";
+import { createRelaySet } from "ndk/utils";
+import { useEffect, useRef, useState } from "react";
+
+export function useFeedWithEose(filter: NDKFilter, relayUrls: string[] = []) {
+  const { ndk } = useNDK();
+  const [feed, setFeed] = useState<NDKEvent[]>([]);
+  const eventBatch = useRef<NDKEvent[] | null>([]);
+
+  useEffect(() => {
+    if (!ndk) {
+      return;
+    }
+
+    const relaySet = createRelaySet(relayUrls, ndk);
+    const subscription = ndk.subscribe(filter, undefined, relaySet);
+
+    subscription.on("event", (event: NDKEvent) => {
+      if (eventBatch.current === null) {
+        setFeed((prev) => [event, ...prev]);
+      } else {
+        eventBatch.current.push(event);
+      }
+    });
+    subscription.on("eose", () => {
+      if (eventBatch.current) {
+        setFeed(eventBatch.current);
+      }
+      eventBatch.current = null;
+    });
+
+    return () => {
+      if (subscription) subscription.stop();
+    };
+  }, [filter, setFeed, ndk]);
+
+  return feed;
+}
```

Comment:
Could this whole component just be an option for useFeed?

Reply:
i.e. `useFeed(filter, { withEose: true });`

Reply:
yeah, but i didn't want to mess with that code for now in case this ended up being a bad idea. easier to just delete this hook if it doesn't work out for the home feed for any reason.

Reply:
gotcha. can always consolidate them later.

Reply:
should we be stopping the previous subscription whenever the filter changes? seems like there may be a memory leak if we don't stop the previous subscription when the filter changes.

Reply:
we do 
```
return () => {
    if (subscription) subscription.stop();
};
```

Reply:
oh nevermind. it's doing that in the cleanup function.